api/v1alpha2/virtualmachine_types.go

```diff
@@ -173,10 +181,30 @@ type VirtualMachineSpec struct {
 
 	// PowerState describes the desired power state of a VirtualMachine.
 	//
+	// Please note this field may be omitted when creating a new VM and will
+	// default to "PoweredOn." However, once the field is set to a non-empty
+	// value, it may no longer be set to an empty value.
+	//
+	// Additionally, setting this value to "Shutdown" or "Suspended" is not
```

Comment:
"Shutdown" is not one of the allowed `VirtualMachinePowerState` values. Maybe was in a prior revision before `VirtualMachinePowerOffMode` was introduced?

One could quickly edit the PowerState before the VM has been reconciled/created on VC so I believe our code still needs to handle a just created-never powered on VM being suspended (I presume by just reporting it as powered off - would want to double check that we don't get into a reconcile error loop).

Reply:
Yes, sorry, I meant to remove these. In an earlier revision there was no PowerOffMode but rather `Shutdown` as one of the CRD power states. I changed it so `kubectl delete vm` could also take advantage of the specified power off mode.